lib/my_array.rb

```diff
@@ -0,0 +1,50 @@
+class MyArray < Array
+  def initialize(array)
+    @array = array
+  end
```

Comment:
Yes, what I hadn't taken into account here is that by inheriting from array, I already had the ability to take in an argument on the .new method.